main.cpp

```diff
@@ -228,346 +305,521 @@ struct University
         std::string hobby = "Play Bass";
         int semestralCredits = 9;
 
-        float weekstudyTime(int research = 28, int teaching = 16);
-        float computeMonthlyExpenses(float food = 250, 
+        float weekStudyTime(int research = 28, int teaching = 16);
+        float computeMonthlyExpenses(float food, 
                                      float fun = 220, 
                                      float other = 350);
     }; 
 
-    //Teach students
-    float teachStudents(Professor Frank); 
-    //Do collaborate research 
-    int doCollaborativeResearch(std::string organization = "UCLA", 
+    float teachStudents(Professor jack);  
+    void doCollaborativeResearch(std::string organization = "UCLA",
                                 std::string department = "Chemistry",
                                 std::string projectName = "influence of fertilizer on plant cultivation"); 
-    //Perform cultural activities
-    int performCulturalActivities(Student Peter,  
+    void performCulturalActivities(Student Peter,  
                                   std::string category = "Arts",
-                                  std::string activity = "Museum Visit"); 
+                                  std::string activity = "Music Museum Visit"); 
 };
 
-/*
-Thing 3) Computer
-5 properties:
-    1) amount of used energy (float)
-    2) amount of required memory to run multiple programs (float)
-    3) amount of daily used disk space (float)
-    4) number of execution errors (int)
-    5) task executed per day (int)
-3 things it can do:
-    1) Execute Programs 
-    2) Save Information
-    3) Connect with other computers
- */
- 
+float University::Professor::work(int research, int teaching)
+{
+    return research + teaching;
+}
+
+float University::Professor::computeMonthlyExpenses(float rent,
+                                                    float food,
+                                                    float fun, 
+                                                    float other)
+{
+    return rent + food + fun + other;
+}
+
+float University::Student::weekStudyTime(int research, int teaching)
+{
+    return research + teaching;
+}
+
+float University::Student::computeMonthlyExpenses(float food = 350,
+                                                  float fun, 
+                                                  float other)
+{
+    return food + fun + other;
+}
+
+float University::teachStudents(Professor) 
```

Comment:
What is the point of declaring a function parameter if you're not going to use it?
- [ ] revise: either use the function parameter or remove the function parameter